code/game/mecha/equipment/mecha_equipment.dm

```diff
@@ -48,14 +48,16 @@
 	return ..()
 
 /obj/item/mecha_parts/mecha_equipment/try_attach_part(mob/user, obj/mecha/M)
-	if(can_attach(M))
-		if(!user.temporarilyRemoveItemFromInventory(src))
-			return FALSE
-		attach(M)
-		user.visible_message("[user] attaches [src] to [M].", "<span class='notice'>You attach [src] to [M].</span>")
-		return TRUE
-	to_chat(user, "<span class='warning'>You are unable to attach [src] to [M]!</span>")
-	return FALSE
+	if(!can_attach(M))
+		to_chat(user, "<span class='warning'>You are unable to attach [src] to [M]!</span>")
+		return FALSE
+	if(!do_mob(user, M, 15))
+		return FALSE
```

Comment:
Should can_attach be checked after this too? Otherwise could get situations where 2 people put something in 1 slot at the same time